includes/Speculative_Prerendering.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Class Speculative_Prerendering
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types = 1);
+
+namespace Google\Web_Stories\Admin;
+
+use Google\Web_Stories\Service_Base;
+
+/**
+ * Speculative_Prerendering class.
+ */
+class Speculative_Prerendering extends Service_Base {
+
+	/**
+	 * Array defining the matches for different pages.
+	 *
+	 * @var array<string,array<string>>
+	 */
+	private array $matches = [
+		'dashboard'   => [ 'post.php?post=*&action=edit', 'post-new.php?post_type=web-story' ],
+		'all_stories' => [ 'post.php?post=*&action=edit', 'post-new.php?post_type=web-story', '/web-stories*' ],
+		'archive'     => [ '/web-stories*' ],
```

Comment:
Maybe we could default to `/web-stories/*` if `get_post_type_archive_link()` returns `false`. I had the archive page disabled, and yet the stories were visible on `/web-stories/*`

Reply:
Ah right, of course, it's for the single views, not the archive. Then let's use at least `Story_Post_Type::REWRITE_SLUG` here instead of hardcoding `web-stories`